### PR TITLE
Add gllvm support

### PR DESCRIPTION
**Pull request by danjujan**
Resolves https://github.com/se-sic/VaRA/issues/711

### CODE REVIEW

**Review by boehmseb — APPROVED** [review]

File: varats-core/varats/experiment/wllvm.py [hunk 1]
```diff
@@ -105,10 +107,17 @@ def __create_distcc_scripts(self) -> tp.Tuple[Path, Path]:
...
 
     def __call__(self, compiler: cc, *args: tp.Any, **kwargs: tp.Any) -> tp.Any:
-        if str(compiler).endswith("clang++"):
-            wllvm = local["wllvm++"]
+        if is_gllvm_available():
+            if str(compiler).endswith("clang++"):
+                wllvm = local["gclang++"]
+            else:
+                wllvm = local["gclang"]
+
```
Comment (vulder) [inline varats-core/varats/experiment/wllvm.py]:
```suggestion
            compiler_wrapper = "gclang"
```

File: varats-core/varats/experiment/wllvm.py [hunk 2]
```diff
@@ -105,10 +107,17 @@ def __create_distcc_scripts(self) -> tp.Tuple[Path, Path]:
...
         else:
-            wllvm = local["wllvm"]
+            if str(compiler).endswith("clang++"):
+                wllvm = local["wllvm++"]
+            else:
+                wllvm = local["wllvm"]
```
Comment (vulder) [inline varats-core/varats/experiment/wllvm.py]:
```suggestion
            compiler_wrapper = "wllvm"

        if str(compiler).endswith("clang++"):
            compiler_wrapper += "++"

        wllvm = local[compiler_wrapper]
```

File: varats-core/varats/experiment/wllvm.py [hunk 3]
```diff
@@ -342,3 +355,9 @@ def get_cached_bc_file_path(
             " compiled with the correct compile/extract action."
         )
     return Path(bc_file_path)
+
+
+def is_gllvm_available() -> bool:
+    return shutil.which("gclang") is not None and shutil.which(
+        "gclang++"
+    ) is not None and shutil.which("get-bc") is not None
```
Comment (vulder) [inline varats-core/varats/experiment/wllvm.py]:
```suggestion
    return None not in { shutil.which(binary) for binary in ["gclang", "gclang++", "get-bc"] }
```

**Review by vulder — CHANGES_REQUESTED** [review]
Otherwise, LGTM

**Review by vulder — APPROVED** [review]

**Pull request merged**